### ISSUE TITLE
Passthrough selected kubeadm flags

### ISSUE THREAD

**Issue by randomvariable** [kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm#268]
/kind feature

**Describe the solution you'd like**
Kubeadm 1.16 introduced `kubeadm init --experimental-kustomize kubeadm-patches/` to allow customization of the static pods for certain environments. Is there a way that  this can be passed through to kubeadm, or allow arbitrary arguments to be appended to kubeadm init/join

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Comment by chuckha** [comment]
Is this supported in any version of the config types?

My other thought is we could consider adding a "kubeadmFlags" section to the CABPK type, but I'd more hope everything on the CLI of kubeadm is exposed in the kubeadm types and we'll get this "for free" as we use the newer kubeadm config types

**Comment by randomvariable** [comment]
It's an experimental flag at the moment, and was introduced for high security environments who want to tailor the static pods delivered by kubeadm. It's the kind of thing that may remain a flag as it's an imperative operation and needs to refer to extra files on disk, which wouldn't survive translation into a ComponentConfig.

**Comment by detiber** [comment]
The one worry that I have with this approach is that adding command line arguments may introduce additional version specific requirements to config beyond which versions support the kubeadm config version we are using. This would massively complicate upgrade scenarios.

**Comment by randomvariable** [comment]
Agreed. Haven't had a specific demand for the future yet, but the main use case this was to enable is switching the control plane so that it doesn't run as root.

Want to work out what the dependency is: Is there some generalisation we can make? Or even place demands on the interface with kubeadm, e.g. a kubeadm feature supported by CAPI can only be done so via ComponentConfig? I can see for example that you'd want to keep the kustomizations applied on upgrade in this case, so the patches would need to be kept somewhere and reapplied.

**Comment by detiber** [comment]
This is definitely tricky...  If we do adopt this feature (as command line args), then I think we'd need to find a way to block upgrades for the cases where it is used to avoid version related command line argument compat issues that are disjoint from the api type support issues.

**Comment by randomvariable** [comment]
Yup. Going to retitle this as "figure out long-term interface for static pod kustomizations with kubeadm". which may or may not also require changes to kubeadm.
